### ISSUE TITLE
training process has started but appears stuck at Step 1 ,  the GPU not being utilized for training

### ISSUE THREAD

**Issue by Johhanliang** [mindverse/Second-Me#189]
2025-04-10 02:02:26 [INFO] train.py:393 - Starting actual training process...
***** Running training *****
Num examples = 107
Num Epochs = 3
Instantaneous batch size per device = 2
Total train batch size (w. parallel, distributed & accumulation) = 2
Gradient Accumulation steps = 1
Total optimization steps = 162
Number of trainable parameters = 14,966,784
2025-04-10 02:02:26 [INFO] train.py:276 - === Training Begin ===
2025-04-10 02:02:26 [INFO] train.py:277 - Checking initial conditions:
0%| | 0/162 [00:00<?, ?it/s]2025-04-10 02:02:26 [INFO] train.py:276 - === Training Begin ===
2025-04-10 02:02:26 [INFO] train.py:277 - Checking initial conditions:
2025-04-10 02:02:26 [INFO] train.py:298 -
=== Starting Step 1 ===
2025-04-10 02:02:26 [INFO] train.py:298 -
2025-04-10 02:02:26 [INFO] train.py:304 - CPU Usage: 0.0%
=== Starting Step 1 ===
2025-04-10 02:02:26 [INFO] train.py:305 - Memory Usage: 878.05MB
2025-04-10 02:02:26 [INFO] train.py:304 - CPU Usage: 0.0%
2025-04-10 02:02:26 [INFO] train.py:308 - Thread Count: 71
2025-04-10 02:02:26 [INFO] train.py:305 - Memory Usage: 878.05MB
2025-04-10 02:02:26 [INFO] train.py:308 - Thread Count: 71
/usr/local/lib/python3.12/site-packages/transformers/trainer.py:3618: FutureWarning: `torch.cpu.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cpu', args...)` instead.
ctx_manager = torch.cpu.amp.autocast(cache_enabled=cache_enabled, dtype=self.amp_dtype)
2025-04-10 02:02:26 [INFO] trainprocess_service.py:834 - Training started
2025-04-10 02:02:26 [INFO] trainprocess_service.py:880 - Progress updated: 0% - Current step: 0/162

**Comment by sudomarcma** [comment]
the default install script only installed torch but never cuda, you have to enable gpu and for docker and install cuda yourself. For now (the current commit 2025/4/10) it will train with cpu only.

**Comment by Johhanliang** [comment]
OK, thank you, but the training process is stuck at step 1 with an overall progress of 80%, and it's not progressing any further. There are no error messages displayed

**Comment by sudomarcma** [comment]
> not progressing any further.

Because it using cpu and it does takes forever. I have mine running right now 1% done👌

**Comment by ovizii** [comment]
I just installed SecondMe and am stuck with the exact same problem. So we do need a GPU as prerequisite? Its not mentioned on the project page. 

Could the training be outsourced i.e. via an API to some service? I am trying to run this on my laptop...

**Comment by yingapple** [comment]
Everyone, our project may not necessarily require a GPU at the moment, but it does need 8GB or more of system memory. This will ensure performance.

**Comment by imaffe** [comment]
> I just installed SecondMe and am stuck with the exact same problem. So we do need a GPU as prerequisite? Its not mentioned on the project page.
> 
> Could the training be outsourced i.e. via an API to some service? I am trying to run this on my laptop...

@ovizii 

I'm using  MBP 32G RAM M2 Pro to train, and it works on the 1.5B model. Maybe you can try rent a cloud GPU machine on AWS or Azure and run Second Me there. But I didn't tried that before.

**Comment by Johhanliang** [comment]
Finally, the model training is done. But then there were issues with merging. So I switched to doing it manually. Anyways, we're at the application part now. The 0.5B model isn't great at answering questions. Maybe the data's no good? The 3B model might be better. I'll test it later.

**Comment by imaffe** [comment]
> Finally, the model training is done. But then there were issues with merging. So I switched to doing it manually. Anyway, the process going to the application part now. The 0.5B model isn't great at answering questions. Maybe the data's no good? The 3B model might be better. I'll test it later.

My experience is that model <= 1.5B is barely usable. Data is definitely important but I didn't feed too many documents into my models. I'm considering moving to a better machine to test >= 7B models.

**Comment by yingapple** [comment]
> Finally, the model training is done. But then there were issues with merging. So I switched to doing it manually. Anyway, the process going to the application part now. The 0.5B model isn't great at answering questions. Maybe the data's no good? The 3B model might be better. I'll test it later.

You’re right. The 0.5B model is just a toy case — a relatively simple example. We’re still aiming for models with 3B parameters or more. You might want to check out the MLX features we recently released — they allow for faster and larger model training on Mac. Of course, we’re using quantized models with QLoRA.

Also you can have a look at our technical report.
